### PR TITLE
Add possibility to print timestamp before pr_* message

### PR DESCRIPTION
**Pull request by danmois**
Add possibility to print timestamp before pr_* message.

### CODE REVIEW

File: include/no_os_delay.h [hunk 1]
```diff
@@ -46,6 +46,11 @@
 
 #include <stdint.h>
 
+struct no_os_time {
+	unsigned int s, us;
+};
+
+
```
Comment (amiclaus) [inline include/no_os_delay.h:53]: nit: extra line

File: include/no_os_print_log.h [hunk 2]
```diff
@@ -42,6 +42,8 @@
 
 #include <stdio.h>
 
+//#define PRINT_TIME
```
Comment (buha) [inline include/no_os_print_log.h:45]: this shouldnt be here

Comment (danmois) [reply include/no_os_print_log.h]: I removed it. It will have to be defined in application configuration files.

**Comment by danmois** [comment]
V2

- remove PRINT_TIME define from no_os_print_log.h file

File: include/no_os_print_log.h [hunk 3]
```diff
@@ -55,54 +55,81 @@
 #define NO_OS_LOG_LEVEL NO_OS_LOG_INFO
 #endif
 
+#if defined(PRINT_TIME)
+#define pr_time			{ struct no_os_time _t = no_os_get_time(); \
+			printf("[%5d.%06d] ", _t.s, _t.us); }
```
Comment (buha) [inline include/no_os_print_log.h:60]: did astyle choose this indentation here ?

**Comment by buha** [comment]
3rd commit should be last, rearrange

**Comment by danmois** [comment]
V3

- adjust indentation for the pr_time macro
- reorder last two commits

File: include/no_os_print_log.h (hunk 3 above)
Comment (danmois) [reply include/no_os_print_log.h]: I modified the indentation and run astyle in the new version.

**Review by buha — APPROVED** [review]
i think it's ok to merge this even if it's not implemented on all platforms

**Pull request merged**